### PR TITLE
Update to use crypto-ld v5.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Propagating changes from upstream PR https://github.com/digitalbazaar/crypto-ld/pull/56.

### CODE REVIEW

File: src/Ed25519VerificationKey2018.js [hunk 1]
```diff
@@ -145,30 +145,25 @@ class Ed25519VerificationKey2018 extends LDVerifierKeyPair {
...
-   * Adds the private key material to the KeyPair.
-   * @param {object} key - A plain object.
-   * @param {string} key.privateKeyBase58 - Base58btc encoded Private Key
-   *
-   * @return {object} The keyNode with encoded private key material.
-   */
-  addPrivateKey({key}) {
-    key.privateKeyBase58 = this.privateKeyBase58;
-    return key;
+  export({publicKey = false, privateKey = false}) {
```
Comment (mattcollier) [inline src/Ed25519VerificationKey2018.js]: Just double checking that the API signature and behavior here is exactly the same was it was before.  Is that the case?  Seems like `publicKey = true` would be a sane default, but if we're keeping existing behavior that's fine.

File: src/Ed25519VerificationKey2018.js [hunk 2]
```diff
@@ -145,30 +145,25 @@ class Ed25519VerificationKey2018 extends LDVerifierKeyPair {
...
+    const exportedKey = super.export({publicKey, privateKey});
```
Comment (mattcollier) [inline src/Ed25519VerificationKey2018.js]: what does this do exactly?  I'm concerned specifically that we're potentially telling `super.export` that we want the `privateKey` for some reason, but then it's being added (again?) here on L164.

Comment (dmitrizagidulin) [reply src/Ed25519VerificationKey2018.js]: Ok, I can see how that's confusing / concerning. I updated it to not use `super.export()` -- see if that makes more sense.

File: src/Ed25519VerificationKey2018.js (hunk 1 above)
Comment (dmitrizagidulin) [reply src/Ed25519VerificationKey2018.js]: That is the previous behavior, yeah; I updated the method to include the existing checks/errors (that were housed in the superclass's method). See if this makes more sense now.

File: src/Ed25519VerificationKey2018.js [hunk 3]
```diff
@@ -145,30 +145,33 @@ class Ed25519VerificationKey2018 extends LDVerifierKeyPair {
   }
 
   /**
-   * Adds a public key base to a public key node.
+   * Exports the serialized representation of the KeyPair
+   * and other information that json-ld Signatures can use to form a proof.
    *
-   * @param {object} key - The public key object in a jsonld-signature.
-   * @param {string} key.publicKeyBase58 - Base58btc encoded Public Key.
+   * @param {object} [options={}] - Options hashmap.
```
Comment (mattcollier) [inline src/Ed25519VerificationKey2018.js:151]: this does not match the API definition.

File: src/Ed25519VerificationKey2018.js [hunk 4]
```diff
@@ -145,30 +145,33 @@ class Ed25519VerificationKey2018 extends LDVerifierKeyPair {
...
-   * @param {object} key - A plain object.
-   * @param {string} key.privateKeyBase58 - Base58btc encoded Private Key
-   *
-   * @return {object} The keyNode with encoded private key material.
-   */
-  addPrivateKey({key}) {
-    key.privateKeyBase58 = this.privateKeyBase58;
-    return key;
+  export({publicKey = false, privateKey = false}) {
+    if(!publicKey && !privateKey) {
```
Comment (mattcollier) [inline src/Ed25519VerificationKey2018.js]: prefer construction 
```suggestion
    if(!(publicKey || privateKey)) {
```

File: src/Ed25519VerificationKey2018.js [hunk 5]
```diff
@@ -145,30 +145,33 @@ class Ed25519VerificationKey2018 extends LDVerifierKeyPair {
...
+      throw new Error(
```
Comment (mattcollier) [inline src/Ed25519VerificationKey2018.js]: I believe this is a `TypeError`  https://nodejs.org/api/errors.html#errors_class_typeerror

File: src/Ed25519VerificationKey2018.js (hunk 3 above)
Comment (dmitrizagidulin) [reply src/Ed25519VerificationKey2018.js:151]: Thanks, fixing.

**Review by mattcollier — APPROVED** [review]
This is more clear, TY.  Just a couple more nits.  Also, personal preference for releasing non-breaking dep updates like this as minor instead of patch.  More of a feature than a fix IMO.

File: src/Ed25519VerificationKey2018.js (hunk 3 above)
Comment (dmitrizagidulin) [reply src/Ed25519VerificationKey2018.js:151]: Fixed (updated the function to match comment)

File: src/Ed25519VerificationKey2018.js (hunk 4 above)
Comment (dmitrizagidulin) [reply src/Ed25519VerificationKey2018.js]: Thanks.

File: src/Ed25519VerificationKey2018.js (hunk 5 above)
Comment (dmitrizagidulin) [reply src/Ed25519VerificationKey2018.js]: Thanks, changed.

**Comment by dmitrizagidulin** [comment]
@mattcollier 
> Also, personal preference for releasing non-breaking dep updates like this as minor instead of patch. More of a feature than a fix IMO.

Np, I also prefer minor version for these. Will change.

File: src/Ed25519VerificationKey2018.js [hunk 6]
```diff
@@ -145,30 +145,33 @@ class Ed25519VerificationKey2018 extends LDVerifierKeyPair {
...
-  }
-
-  /**
-   * Adds the private key material to the KeyPair.
-   * @param {object} key - A plain object.
-   * @param {string} key.privateKeyBase58 - Base58btc encoded Private Key
-   *
-   * @return {object} The keyNode with encoded private key material.
-   */
-  addPrivateKey({key}) {
```
Comment (dlongley) [inline src/Ed25519VerificationKey2018.js:169]: I don't understand how this isn't a breaking change. Can you explain?

Comment (dlongley) [reply src/Ed25519VerificationKey2018.js:169]: So I've been told these were private functions that weren't properly marked as such.

Comment (dmitrizagidulin) [reply src/Ed25519VerificationKey2018.js:169]: @dlongley - those two functions were not really public API. Those were helper functions (just without the `_` in the name). There's no reason for anybody to have used them standalone, they don't do anything outside of `export()`.

**Review by dlongley — APPROVED** [review]
So doing a minor release for this is a little iffy. The argument is that the functions removed were private and not part of the public documentation on the README -- and it was just a mistake to not mark them with an underscore. I would like to let this go through, however, I don't have any data on how likely it is that anyone used those functions so... I don't think that's good enough to do a shaky minor release.

I'll approve with the caveat that it becomes a major release or that I'm overruled by the other maintainers thinking it's ok to proceed with a minor release here.

**Comment by dmitrizagidulin** [comment]
@dlongley - no prob. Changed it to a breaking change, in the changelog.

**Comment by mattcollier** [comment]
This seems like a good time to mention REAL private class methods and fields: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Available in Node 12+.

**Comment by dmitrizagidulin** [comment]
@mattcollier - re private fields -- oh, fascinating! Doesn't seem like there's good coverage in the browsers, though. So maybe we should wait?

**Pull request merged**